api/health.go

```diff
@@ -318,15 +367,17 @@ func GetUnitsProperties(config *Config) (UnitsHealthResponseJSONStruct, error) {
 	// after we finished querying systemd units, close dbus connection
 	if err = config.DcosTools.CloseDbusConnection(); err != nil {
 		// we should probably return here, since we cannot guarantee that all units have been queried.
-		return UnitsHealthResponseJSONStruct{}, err
-	}
-	return UnitsHealthResponseJSONStruct{
-		Array:       allUnitsProperties,
-		Hostname:    config.DcosTools.GetHostname(),
-		IPAddress:   config.DcosTools.DetectIP(),
-		DcosVersion: config.DcosVersion,
-		Role:        config.DcosTools.GetNodeRole(),
-		MesosID:     config.DcosTools.GetMesosNodeID(config.DcosTools.GetNodeRole(), "id"),
-		TdtVersion:  config.Version,
-	}, nil
+		return healthReport, err
+	}
+
+	// update the rest of healthReport fields
+	healthReport.Array = allUnitsProperties
+	healthReport.Hostname = config.DcosTools.GetHostname()
+	healthReport.IPAddress = config.DcosTools.DetectIP()
```

Comment:
👍 I like the `.DcosTools` nameing convention. Might want to consider `DCOSTools` following general go linter guidelines.